metadata/src/lib.rs

```diff
@@ -214,7 +214,10 @@ impl Metadata {
         MetadataHasher::new(self)
     }
 
-    /// Filter out any pallets that we don't want to keep, retaining only those that we do.
+    /// Filter out any pallets and/or runtime_apis that we don't want to keep, retaining only those that we do.
+    /// Note:
+    /// without also filtering out irrelevant runtime_api's in addition to pallets one should not expect significant metadata size reduction.
```

Comment:
```suggestion
    /// runtime api filtering will not lead to significant metadata size reduction because the return types are kept to ensure that those can be decoded.
```